### ISSUE TITLE
Container having issue publishing logs

### ISSUE THREAD

**Issue by alen-z** [mitmproxy/mitmproxy#5727]
#### Problem Description
Running container image using Docker in non-interactive mode or in Kubernetes does not show logs.

#### Steps to reproduce the behavior:
1. `docker run --rm -p 8080:8080 -p 8081:8081 mitmproxy/mitmproxy:9.0.1 mitmweb --web-host 0.0.0.0`
2. Not able to see STDOUT or exit the container with `Ctrl-c`.
3. Only `docker kill` works.

On the other hand `docker run -it ...` works.

`docker run --rm -p 8080:8080 -p 8081:8081 mitmproxy/mitmproxy:9.0.1 gosu mitmproxy mitmweb --web-host 0.0.0.0` does not show logs, but allows the exit with `Ctrl-c`

#### Expected behavior
`docker run --rm -p 8080:8080 -p 8081:8081 mitmproxy/mitmproxy:9.0.1 mitmweb --web-host 0.0.0.0` prints logs real-time and can be stopped with `Ctrl-c`.

#### System Information
Using latest 9.0.1 container image.

**Comment by CraftyCanine** [comment]
FYI if anyone stumbles on this, running with docker compose can be achieved by using the stdin_open: true and tty: true options.
https://stackoverflow.com/questions/36249744/interactive-shell-using-docker-compose

**Comment by alen-z** [comment]
Yes, similar how it's done here if you look at values file: https://artifacthub.io/packages/helm/mitmproxy/mitmproxy
